include/internal/refcount.h

```diff
@@ -16,26 +16,32 @@
 #  endif
 # endif
 
+# ifndef CRYPTO_REF_TYPE
```

Comment:
It would be a cleaner if this file wasn't being dual purposed via some preprocessor magic.  Contaminating the _refcount_ code seems wrong to me.

I think Kurt's suggestion above is cleaner, nicer and less invasive.  Especially for something that is likely to go away with #6790